applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.h

```diff
@@ -321,19 +321,41 @@ class TreeContactSearch
     
     void CreatePointListMortar()
     {
+        // Clearing the vector
+        mPointListDestination.clear();
+        
         // Iterate in the conditions
         ConditionsArrayType& conditions_array = mrMainModelPart.Conditions();
         const int num_conditions = static_cast<int>(conditions_array.size());
 
-        #pragma omp for nowait schedule(static)
-        for(int i = 0; i < num_conditions; i++) 
+        // Creating a buffer for parallel vector fill
+        const unsigned int num_threads = omp_get_max_threads();
+        std::vector<PointVector> points_buffer(num_threads);
+
+        #pragma omp parallel
         {
-            auto it_cond = conditions_array.begin() + i;
+            const unsigned int Id = omp_get_thread_num();
```

Comment:
call this "thread_id" or "processor_id" to make it clear that it is not the Id of a kratos entity